katsdpdata/prod_handler.py

```diff
@@ -1,53 +1,501 @@
-import logging
-import socket
+#!/usr/bin/env python3
+
+"""Parallel file uploader to trawl NPY files into S3."""
 
 import boto
 import boto.s3.connection
+import concurrent.futures as futures
+import multiprocessing
+import os
+import re
+import shutil
+import time
+from pathlib import PurePath
+
+from katsdpdata.met_detectors import file_type_detection
+from katsdpdata.met_extractors import MetExtractorException
+from katsdpdata.met_handler import MetaDataHandler
+from katsdpdata.utilities import get_s3_connection
+from katsdpdata.utilities import redact_key
+from katsdpdata.utilities import s3_create_anon_access_policy
+
+CAPTURE_BLOCK_REGEX = "^[0-9]{10}$"
+CAPTURE_STREAM_L0_REGEX = "^[0-9]{10}[-_].*l0$"
+CAPTURE_STREAM_L1_REGEX = "^[0-9]{10}[-_].*l1-flags$"
```

Comment:
Personally I put configuration in a config.py and then import them into other modules. I have been burnt too many times having the same thing defined in multiple places and then getting out of sync. I also normally allow the defaults to be overwritten from the environment. Helps with testing and deploying as you can just put all of these into a .env file and load that with a docker-compose. If you wanted to try out different configurations in the future, it is much easier to change a .env file in the katsdpinfrastructure repo and redeploy than having to update the codebase, build a new docker, deploy the new docker. Also allows you to set credentials using .env (which I always have in gitignore) then you don't mistakenly end up uploading secrets.

Reply:
Hived out into JIRA ticket.